echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/api/Notification.groovy

```diff
@@ -50,4 +54,40 @@ class Notification {
     NORMAL,
     HIGH
   }
+
+  /**
+   * Allows Spinnaker services sending Notifications through Echo to specify one or more interactive actions
+   * that, when acted upon by a user, cause a callback to Echo which gets routed to that originating service.
+   */
+  static class InteractiveActions {
```

Comment:
The classes here provide the generic framework for Spinnaker services to generate interactive notifications that are not specific to any implementation (Slack, e-mail, whatever). Currently supporting only buttons, but we could add whatever we want/need to the framework.